astropy/cosmology/core.py

```diff
@@ -594,11 +599,11 @@ def __equiv__(self, other):
         # Check if have equivalent parameters and all parameters in `other`
         # match those in `self`` and `other`` has no extra parameters.
         params_eq = (
-            set(self.__all_parameters__) == set(other.__all_parameters__)  # no extra
+            # no extra parameters
+            self._parameters_all == other._parameters_all
             # equal
             and all(
-                np.all(getattr(self, k) == getattr(other, k))
-                for k in self.__parameters__
+                np.all(getattr(self, k) == getattr(other, k)) for k in self.parameters
```

Comment:
Same

Reply:
The use of `np.all` is required, e.g. for `m_nu`, so changing to `==` won't work.

Reply:
But... it looks like `np.equal` handles this correctly!

Reply:
But `np.equal` is unhappy about this statically. It looks like this is an unintentional use case of `np.equal` and I'm not confident of it sticking around.

Reply:
Equality would work if `m_nu` were a structured array, but there is no real reason for it being that. For now, just keep things as you had them (but perhaps add a comment).

Reply:
Done.

Reply:
Added a comment. Resolving this thread.